src/devices/GoPiGo3/Movements/Motor.cs

```diff
@@ -0,0 +1,203 @@
+        /// Create a motor
+        /// </summary>
+        /// <param name="port">Motor port</param>
+        public Motor(GoPiGo brick, MotorPort port) : this(brick, port, 1000) { }
+
+        public Motor(GoPiGo brick, MotorPort port, int timeout)
+        {
+            if (port == MotorPort.Both)
+                throw new ArgumentException($"Motor class can only have 1 motor");
+            _goPiGo = brick;
+            Port = port;
+            _periodRefresh = timeout;
+            _timer = new Timer(UpdateSensor, this, TimeSpan.FromMilliseconds(timeout), TimeSpan.FromMilliseconds(timeout));
+        }
+
+        /// <summary>
+        /// Set the speed of the motor
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void SetSpeed(int speed)
+        {
+            speed = Math.Clamp(speed, -255, 255);
+            _goPiGo.SetMotorPower(Port, speed);
+        }
+
+        /// <summary>
+        /// Set Tachometer encoder offset
+        /// Use this to reset or setup a specific position
+        /// </summary>
+        /// <param name="position">New offset, 0 to reset</param>
+        public void SetTachoCount(int position)
+        {
+            _goPiGo.SetMotorPosition(Port, position);
+        }
+
+        /// <summary>
+        /// Stop the Motor
+        /// </summary>
+        public void Stop()
+        {
+            _goPiGo.SetMotorPower(Port, 0);
+        }
+
+        /// <summary>
+        /// Start the motor
+        /// </summary>
+        public void Start()
+        {
+            _goPiGo.SetMotorPower(Port, Speed);
+        }
+
+        /// <summary>
+        /// Start with the specified speed
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void Start(int speed)
+        {
+            SetSpeed(speed);
+            Start();
+        }
+
+        /// <summary>
+        /// Change the polatity of the motor
+        /// </summary>
+        /// <param name="polarity">Polarity of the motor, backward, forward or opposite</param>
+        public void SetPolarity(Polarity polarity)
+        {
+            try
+            {
+                var motorstatus = _goPiGo.GetMotorStatus(Port);
+                switch (polarity)
+                {
+                    case Polarity.Backward:
+                        if (motorstatus.Speed > 0)
+                            _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    case Polarity.Forward:
+                        if (motorstatus.Speed < 0)
+                            _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    case Polarity.OppositeDirection:
+                        _goPiGo.SetMotorPower(Port, -Speed);
+                        break;
+                    default:
+                        break;
+                }
+            }
+            catch (Exception) { }
```

Comment:
Why are we catching all potential exceptions rather than surfacing them to the user?

Reply:
I reduced to the IOException only. The point is that you may get error in communicating with the shield, that happens time to time. Rather than blocking everything, it seems to work better by just ignoring them for some of the motors. Movement, that's at least my experience. Now, very open to remove it and leave it to the user to manage.